src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -829,108 +830,132 @@ subroutine init_atm_static(mesh, dims, configs)
-             ii = (i - 1) / supersample_fac + 1
-             jj = (j - 1) / supersample_fac + 1
+        do while (.not. mgr % is_stack_empty())
+            tile => mgr % pop_tile()
 
-             lat_pt = start_lat - (supersample_fac*(jTileStart-1) + j - (supersample_fac*3+1)) * 0.05 / supersample_fac
-             lon_pt = start_lon + (supersample_fac*(iTileStart-1) + i - (supersample_fac*3+1)) * 0.05 / supersample_fac
-             lat_pt = lat_pt * PI / 180.0
-             lon_pt = lon_pt * PI / 180.0
+            if (tile % is_processed) then
+                cycle
+            end if
 
-             iPoint = nearest_cell(lat_pt,lon_pt,iPoint,nCells,maxEdges, &
-                                   nEdgesOnCell,cellsOnCell, &
-                                   latCell,lonCell)
-             if (rarray(ii,jj,1) /= msgval) then
+            call mpas_log_write('Processing tile: '//trim(tile % fname))
+
+            all_pixels_mapped_to_halo_cells = .true.
+
+            do j = supersample_fac * tile_bdr + 1, supersample_fac * (tile_ny + tile_bdr), 1
+                do i = supersample_fac * tile_bdr + 1, supersample_fac * (tile_nx + tile_bdr), 1
+
+                    ii = (i - 1) / supersample_fac + 1
+                    jj = (j - 1) / supersample_fac + 1
+
+                    call mgr % tile_to_latlon(tile, j, i, lat_pt, lon_pt, supersample_fac)
+                    call mpas_latlon_to_xyz(xPixel, yPixel, zPixel, sphere_radius, lat_pt, lon_pt)
+                    call mpas_kd_search(tree, (/xPixel, yPixel, zPixel/), res)
+
+                    if (tile % tile(ii, jj, 1) * scalefactor /= missing_value) then
+                        !
+                        ! This field only matters for land cells, and for all but the outermost boundary cells,
+                        ! we can safely assume that the nearest model grid cell contains the pixel (else, a different
+                        ! cell would be nearest)
+                        !
+                        if (landmask(res % id) == 1 .and. bdyMaskCell(res % id) < nBdyLayers) then
+                            snoalb(res % id) = snoalb(res % id) + tile % tile(ii, jj, 1)
+                            nhs(res % id) = nhs(res % id) + 1
+                            if (res % id <= nCellsSolve) then
+                                all_pixels_mapped_to_halo_cells = .false.
+                            end if
+                        ! For outermost land cells, additional work is needed to verify that the pixel
+                        ! actually lies within the nearest cell
+                        else if (landmask(res % id) == 1) then
+                             if (mpas_in_cell(xPixel, yPixel, zPixel, xCell(res % id), yCell(res % id), zCell(res % id), &
+                                              nEdgesOnCell(res % id), verticesOnCell(:,res % id), xVertex, yVertex, zVertex)) then
+
+                                    snoalb(res % id) = snoalb(res % id) + tile % tile(ii, jj, 1)
```

Comment:
For the interpolation of terrain, we performed the accumulation in a 64-bit integer array to avoid order-of-summation differences when running with different MPI task counts. Should we adopt a similar approach here? In single-precision, we can represent integral values up to about 2^24 (23 mantissa bits, plus an implicit 1 bit, I think), or 16777216, without having any gaps in the range of values. With the albedo values having up to four significant digits (up to 100 percent, with two decimal places of precision), it seems like we could run into issues when we accumulate around ~1700 pixels in each grid cell.

Reply:
Yes that sounds like a good idea. I've just pushed a branch to accumulate values using an integer as you described.